pkg/eventsapi/event_container.go

```diff
@@ -0,0 +1,30 @@
+package eventsapi
+
+import (
+	"encoding/json"
+)
+
+type EventContainer struct {
+	EventVersion EventVersion
+	EventData    json.RawMessage
+}
+
+func (ec *EventContainer) UnmarshalEvent() (Event, error) {
+	jsonEventData, err := ec.EventData.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	switch ec.EventVersion {
+	case EventVersion1:
+		var v1Event EventV1
```

Comment:
nit: we can use shorter names for these local variables like `v1` or `event`.

Reply:
Fixed here: 989d021 :)

Reply:
Thank you!